examples/md-flexible/tests/domainDecomposition/MixedBoundaryConditionTest.cpp

```diff
@@ -86,14 +96,24 @@ TEST_F(MixedBoundaryConditionTest, testSimpleReflection) {
 
   EXPECT_NO_THROW(domainDecomposition.reflectParticlesAtBoundaries(autoPasContainer));
 
-  std::vector<std::array<double, 3>> expectedVelocities = {
-      {0.0,1.0,1.0}, {0.0,1.0,1.0}, {0.0,1.0,0.5}, {0.0,1.0,0.5}, {1.0,0.5,0.2},
-      {0.0,-1.0,-1.0}, {0.0,-1.0,-1.0}, {0.0,-1.0,-0.5}, {0.0,-1.0,-0.5}, {-1.0,-0.5,-0.2}
-  };
+  // derive the expected velocities by reflecting the velocities in the appropriate directions
+  std::vector<std::array<double, 3>> expectedVelocities = particleVelocities;
+  expectedVelocities[0][1] *= -1;
+  // particle 1 should not reflect
+  expectedVelocities[2][1] *= -1; expectedVelocities[2][2] *= -1;
+  expectedVelocities[3][1] *= -1;
+  expectedVelocities[4][1] *= -1; expectedVelocities[4][2] *= -1;
+  expectedVelocities[5][1] *= -1;
+  // particle 6 should not reflect
+  expectedVelocities[7][1] *= -1; expectedVelocities[7][2] *= -1;
+  expectedVelocities[8][1] *= -1;
+  expectedVelocities[9][1] *= -1; expectedVelocities[9][2] *= -1;
+
+
 
   for (auto particle = autoPasContainer->begin(autopas::IteratorBehavior::owned); particle.isValid(); ++particle) {
     const auto id = particle->getID();
-    const auto velocity = particle->getV();
+    const auto &velocity = particle->getV();
     EXPECT_NEAR(velocity[0], expectedVelocities[id][0],1e-13);
     EXPECT_NEAR(velocity[1], expectedVelocities[id][1],1e-13);
     EXPECT_NEAR(velocity[2], expectedVelocities[id][2],1e-13);
```

Comment:
if you don't do the value parametrized tests add more output so that it is easier to understand which particle is affected if something fails here:
```suggestion
    EXPECT_NEAR(velocity[0], expectedVelocities[id][0],1e-13) << "For particle " << id;
    EXPECT_NEAR(velocity[1], expectedVelocities[id][1],1e-13) << "For particle " << id;
    EXPECT_NEAR(velocity[2], expectedVelocities[id][2],1e-13) << "For particle " << id;
```

Reply:
Done this - in the future will use value parametrised tests though